src/app/digital/utils/ExpressionParser/GenerateTokens.ts

```diff
@@ -0,0 +1,93 @@
+    let endIndex = index + 1;
+    while(endIndex < expression.length) {
+        for(const op of OpsArray) {
+            if (SubEquals(expression, endIndex, ops.get(op)))
+                return {type: "input", name: expression.substring(index, endIndex)};
+        }
+        if (SubEquals(expression, endIndex, ops.get("separator")))
+            return {type: "input", name: expression.substring(index, endIndex)};
+        endIndex++;
+    }
+    return {type: "input", name: expression.substring(index, endIndex)};
+}
+
+/**
+ * Gets a token from a given expression starting at a certain index
+ * 
+ * @param expression the expression to extract the token from
+ * @param index the index where the token starts
+ * @param ops the representation format for the operations used in this expression
+ * @returns the token extracted from the expression or null if the index points to the starting location of
+ *              a separator (like " ")
+ */
+function getToken(expression: string, index: number, ops: Map<FormatLabels, string>): Token | null {
+    for(const op of OpsArray) {
+        if (SubEquals(expression, index, ops.get(op)))
+            return {type: op};
+    }
+    if (SubEquals(expression, index, ops.get("separator")))
+        return null;
+    return getInput(expression, index, ops);
+}
+
+/**
+ * Converts the given expression to an array of tokens
+ * 
+ * @param expression the expression to convert
+ * @param ops the representation format for the operations used in this expression
+ * @returns a list of tokens that represent the given expression
+ * @throws {Error} if ops is missing the keys "|", "^", "&", "(", ")", or "separator"
+ * @throws {Error} if the value in ops for keys "|", "^", "&", "(", ")", or "separator" is ""
+ */
+export function GenerateTokens(expression: string, ops: Map<FormatLabels, string>): Token[] {
+    for(const op of OpsArray) {
+        if (!ops.has(op))
+            throw new Error("No " + op + " in supplied operation symbols");
+        if (ops.get(op) === "")
+            throw new Error("Length zero " + op + " in supplied operation symbols");
+    }
+    if (!ops.has("separator"))
+        throw new Error("No separator in supplied operation symbols");
+    if (ops.get("separator") === "")
+        throw new Error("Length zero separator in supplied operation symbols");
+
+    const tokenList = new Array<Token>();
+    let extraSkip = 0;
+    let token: Token;
+
+    let index = 0;
+
+    while(index < expression.length) {
+        extraSkip = 0;
+
+        token = getToken(expression, index, ops);
+        if (token === null)
+            index += ops.get("separator").length;
+        else if (token.type === "input") {
+            tokenList.push(token);
+            index += token.name.length;
+        }
+        else {
+            tokenList.push(token);
+            index += ops.get(token.type).length;
+        }
```

Comment:
nit: don't mix and match one-liner no-brace-usage with else-ifs that have braces, just have braces for each statement

nit: end of if `}` can have the elseif/else on the same line